main.tf

```diff
@@ -49,7 +49,7 @@ locals {
 }
 
 resource "ibm_iam_authorization_policy" "scc_cos_s2s_access" {
-  count                       = var.skip_cos_iam_authorization_policy ? 0 : 1
+  count                       = var.existing_scc_instance_crn != null || var.skip_cos_iam_authorization_policy ? 0 : 1
```

Comment:
I think you should update the variable description of `skip_cos_iam_authorization_policy` to explicitly call out that the value is ignored if passing an existing SCC instance.